jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/dao/AudioPluginDaoTest.kt

```diff
@@ -0,0 +1,156 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Orature.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package org.wycliffeassociates.otter.jvm.workbookapp.persistence.dao
+
+import org.jooq.exception.DataAccessException
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.wycliffeassociates.otter.jvm.workbookapp.persistence.database.AppDatabase
+import org.wycliffeassociates.otter.jvm.workbookapp.persistence.database.InsertionException
+import org.wycliffeassociates.otter.jvm.workbookapp.persistence.entities.AudioPluginEntity
+import java.io.File
+
+class AudioPluginDaoTest {
+    private val testDatabaseFile = File.createTempFile(
+        "test-audio-plugin-dao", ".sqlite"
+    ).also(File::deleteOnExit)
+    private lateinit var database: AppDatabase
+    private val dao by lazy { database.audioPluginDao }
+
+    private val defaultEnity = AudioPluginEntity(
+        id = 0,
+        name = "test",
+        version = "test",
+        bin = "test-jar",
+        args = "test",
+        edit = 0,
+        record = 0,
+        mark = 0,
+        path = null
+    )
+
+    @Before
+    fun setup() {
+        database = AppDatabase(testDatabaseFile)
+    }
+
+    @Test
+    fun testInsert() {
+        Assert.assertEquals(0, dao.fetchAll().size)
+        val id1 = dao.insert(defaultEnity)
+        Assert.assertEquals(1, dao.fetchAll().size)
+
+        // insert the same entity will trigger an update
+        val id2 = dao.insert(defaultEnity)
+
+        Assert.assertEquals(1, dao.fetchAll().size)
+        Assert.assertEquals(id1, id2)
+    }
+
+    @Test
+    fun testInsertThrowsException() {
+        try {
+            dao.insert(defaultEnity.copy(id = 1))
+            Assert.fail("Inserting nonzero id should throw an exception")
+        } catch (e: InsertionException) { }
+    }
+
+    @Test
+    fun testFetchById() {
+        val insertedId = dao.insert(defaultEnity)
+        val entity = dao.fetchById(insertedId)
+
+        Assert.assertNotNull(entity)
+        Assert.assertEquals(
+            defaultEnity.copy(id = insertedId),
+            entity
+        )
+    }
+
+    @Test
+    fun testUpdate() {
+        val insertedId = dao.insert(defaultEnity)
+        val updated = AudioPluginEntity(
+            id = insertedId,
+            name = "updated",
+            version = "updated",
+            bin = "updated",
+            args = "updated",
+            edit = 1,
+            record = 1,
+            mark = 1,
+            path = "updated"
+        )
+
+        dao.update(updated)
+
+        Assert.assertEquals(
+            updated,
+            dao.fetchById(insertedId)
+        )
+    }
+
+    @Test
+    fun `test update duplicated values throws exception`() {
```

Comment:
I like how descriptive this function name is, but it's inconsistent with the other test functions in this class.  For readability and maintainability, the test function names should follow the same pattern.